pkg/server/admin.go

```diff
@@ -2220,7 +2220,9 @@ func (s *adminServer) DecommissionStatus(
 	for nodeID := range replicaCounts {
 		l, ok := s.server.nodeLiveness.GetLiveness(nodeID)
 		if !ok {
-			return nil, errors.Newf("unable to get liveness for %d", nodeID)
+			// Decommissioning an unknown node is ok, but we don't have an appropriate
```

Comment:
This makes me nervous. This is pulling from the executing node's in-mem view of liveness, which is not guaranteed to be complete. Do we really want to silently return incomplete data in that case? I think we need to go through KV here so that we're more confident that the nodeIDs are unused, but also I would still strongly prefer returning something here, something like `{ReplicaCount: 0 Membership: -1}` (unfortunately we don't have an "unknown" value available in the enum).

A maybe even cleaner option would be to return a structured error instead of the one prior to this commit (but keep checking KV as source of truth). Then the recipient can short-circuit to doing just the cleanup right there and then.

Reply:
Yeah, the problem with any API/client changes is that it wouldn't be backportable, so THG would have to wait for 22.1. Going through KV seems reasonable though.

Reply:
>I would still strongly prefer returning something here, something like {ReplicaCount: 0 Membership: -1} (unfortunately we don't have an "unknown" value available in the enum).

Btw, when Protobuf (i.e. the client) receives an unknown enum value, it is set to the zero value. Which in this case means that an older version of the CLI would get `ACTIVE` instead of `-1`, which would cause the decommission command to hang forever.